dbt/contracts/graph/parsed.py

```diff
@@ -405,6 +485,33 @@ def find_operation_by_name(self, name, package):
         return self._find_by_name(name, package, 'macros',
                                   [NodeType.Operation])
 
+    def add_nodes(self, new_nodes):
+        """Add the given dict of new nodes to the manifest."""
+        for unique_id, node in new_nodes.items():
+            if unique_id in self.nodes:
+                raise_duplicate_resource_name(node, self.nodes[unique_id])
+            self.nodes[unique_id] = node
+
+    def patch_nodes(self, patches):
+        """Patch nodes with the given dict of patches. Note that this consumes
+        the input!
+        """
+        # because we don't have any mapping from node _names_ to nodes, and we
+        # only have the node name in the patch, we have to iterate over all the
+        # nodes looking for matching names. We could use _find_by_name if we
+        # were ok with doing an O(n*m) search (one nodes scan per patch)
+        for node in self.nodes.values():
+            if node.resource_type != NodeType.Model:
+                continue
+            patch = patches.pop(node.name, None)
+            if not patch:
+                continue
+            node.patch(patch)
+
+        # if we have any left, some references could not be resolved
+        if patches:
+            raise_patch_targets_not_found(patches)
```

Comment:
would this raise if a schema.yml entry is provided for a nonexistent model? In the v1 schema world, we just treat this as a warning. The idea is that you might disable a particular model with `config(enabled=false)`. If tests exist for that model, it would be unfortunate if it bricked your whole run. I may be way off here, but just wanted to flag it

Reply:
Yeah - we're currently not including disabled models in the graph. We did this because we kept finding bugs where we neglected to check if a model was disabled or not. I'm open to handling this differently in the future

Reply:
Should we at least log a warning or anything, or just ignore this completely?

Reply:
We currently log a message like this to the debug logs:
```
2018-07-26 10:37:24,414 (MainThread): WARNING: Model 'test.my_project.unique_disabled_id' depends on model 'disabled' which was not found or is disabled
```

I think we can probably just do the same thing for the v2 flow